ledger/participant-integration-api/src/main/scala/platform/store/interning/StringInterningCache.scala

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.interning
+
+import com.daml.dec.DirectExecutionContext
+import com.daml.lf.data.Ref
+import com.daml.logging.LoggingContext
+
+import scala.concurrent.Future
+
+class DomainStringIterators(
+    val parties: Iterator[String],
```

Comment:
Any specific reason why these are `Iterators` ? When would we want to have a lazy struct passed in here? Reminds me of the issue with the byte transport issue that we had in the Jdbc layer and it seems that this type is inviting to it.

Reply:
not to construct intermediate datastructures